README.md

```diff
@@ -1,3 +1,270 @@
+
+
+Dials is an extensible configuration package for Go.
+
+## Installation
+
+```
+go get github.com/vimeo/dials
+```
+
+## Prerequisites
+
+Dials requires Go 1.13 or later.
+
+## What is Dials
+
+Dials is a configuration package for Go applications. It supports several different configuration sources including:
+ * JSON, YAML, and TOML config files
+ * environment variables
+ * command line flags (for both Go's [flag](https://golang.org/pkg/flag) package and [pflag](https://pkg.go.dev/github.com/spf13/pflag) package)
+ * watched config files and re-reading when there are changes to the watched files
+ * default values
+
+## Why choose Dials
+Dials is a configuration solution that supports several configuration sources so you only have to focus on the business logic.
+Define the configuration struct and select the configuration sources and Dials will do the rest. Dials is designed to be extensible so if the built-in sources don't meet your needs, you can write your own and still get all the other benefits. Moreover, setting defaults doesn't require additional function calls.
+Just populate the config struct with the default values and pass the struct to Dials. 
+Dials also allows the flexibility to choose the precedence order to determine which sources can overwrite the configuration values. Additionally, Dials has special handling of structs that implement [`encoding.TextUnmarshaler`](https://golang.org/pkg/encoding/#TextUnmarshaler) so structs (like [`IP`](https://pkg.go.dev/net?tab=doc#IP) and [`time`](https://pkg.go.dev/time?tab=doc#Time)) can be properly parsed.
+
+## Using Dials
+
+### Reading from config files, environment variables, and command line flags
+Dials requires very minimal configuration to populate values from several sources through the `ez` package. Define the config struct and provide a method `ConfigPath() (string, bool)` to indicate the path to the config file. The package defines several functions that help populate the config struct by reading from config files,
+environment variables, and command line flags
+
+``` go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vimeo/dials/ez"
+)
+
+// Config is the configuration struct needed for the application
+type Config struct {
+	// When a struct tag more specifically corresponding to a source is
+	// present, it takes precedence over the `dials` tag. Note that just
+	// because there is a `yaml` struct tag present doesn't mean that other
+	// sources can't fill this field.
+	Val1 string `dials:"Val1" yaml:"b"`
+	// the dials tag can be used as an alias so when the name in the config file
+	// changes, the code doesn't have to change.
+	Val2 int `dials:"val_2"`
+	// Dials follows the Go convention for flags and will look for the dials
+	// tag or field name in lower-kebab-case. Without any struct tags, it
+	// would look for val-3 flag. To specify a different flag name, use the
```

Comment:
Hmm, Go doesn't have a particularly strong flag-naming convention. Either kebab or snake case are common.

Also, I wouldn't describe it as dial will "look for" a flag, so much as, it will "register a flag" with that name. (as you know, the flag and pflag sources iterate over set flags)